examples/bps.rst

```diff
@@ -28,8 +28,8 @@ regular :math:`u`, :math:`v`, and :math:`f`, we have:
 
 .. math::
    \begin{aligned}
-   \langle v,u \rangle &:= \int_{\Omega} \, v \, u \, dV ,\\
-   \langle v,f \rangle &:= \int_{\Omega} \, v \, f \, dV .
+   \langle u,v \rangle &:= \int_{\Omega} \, u \, v \, dV ,\\
+   \langle f,v \rangle &:= \int_{\Omega} \, f \, v \, dV .
```

Comment:
Is there a reason for this notation change? I prefer the `(v, u)` ordering, corresponding to the complex conjugation `\int v^* u`. This ordering is more common in physics and engineering, and relates bilinear forms to matrices as `a(v, u) ~ v^T A u`.

The other convention pops up quite a bit in the math literature, but I think it takes more effort to relate to discrete operators and track conjugation/transposition.

Reply:
I was just trying to be consistent with the definition given above on line 21 of the `FEMtheory.rst.inc` file, where I had specified that that was the definition of the inner product for real-valued functions. 
FWIW, I am not sure I've seen many times the notation you mention, of defining `(v,u) = \int v^* u `. I've seen more often the definition as `(u,v) = \int v^* u `, so with the functions in the integrand swapped, not the ones in the bilinear form operator. This also matches the notation for the complex conjugate of an inner product, i.e., `\langle x, y \rangle = \bar{ \langle y, x \rangle }`, or the complex conjugate of the product of matrices, i.e., `(AB)^* = B^* A^*`

Reply:
Yeah, it's more common in physics/engineering convention. I typically use it specifically because it maps more directly to the matrix expressions.
> Some authors, especially in physics and matrix algebra, prefer to define the inner product and the sesquilinear form with linearity in the second argument rather than the first. ...
> https://en.wikipedia.org/wiki/Inner_product_space#Alternative_definitions,_notations_and_remarks

> The inner product on Hilbert space (   ,   ) {\displaystyle (\ ,\ )} {\displaystyle (\ ,\ )} (**with the first argument anti linear as preferred by physicists**) ...
> https://en.wikipedia.org/wiki/Bra%E2%80%93ket_notation

https://www.netlib.org/lapack/explore-html/d2/df9/group__complex16__blas__level1_ga45b00bad9285ff50cd86e97dfb04facd.html#ga45b00bad9285ff50cd86e97dfb04facd

It is true that PETSc chose the more common math convention for VecDot.
https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Vec/VecDot.html

Reply:
I would personally stick to the most common mathematical notation of `(first, second)` with the second argument being conjugate linear, rather than the first one. This embraces both real-valued and complex-valued cases. In fact, in the case of real-valued vectors/functions the order of the operands on the RHS doesn't really matter. For the complex-valued case, it resembles the matrix form: `\langle x, y \rangle = y^+ M x`. 
In the eventuality we wanted to use the bra-ket notation, we should remark that 
⟨·, ·⟩ and ⟨· | ·⟩ are distinct notations. But honestly since we don't even have quantum mechanics examples for now, I would not adopt a niche notation in that field for general-purpose documentation that we have right now.

Reply:
Are you saying that the resemblance of the first is more natural than the second?
* `(A x, B y) ~ y^* (B^* A) x`
* `(A x, B y) ~ x^* (A^* B) y`

FWIW, it isn't a quirk in quantum mechanics, but also widespread in engineering literature; e.g., Hughes and associates:
![Screenshot_20200402_114422](https://user-images.githubusercontent.com/3303/78281030-486b3e80-74d7-11ea-84eb-f5915ecebb51.png)
![Screenshot_20200402_114445](https://user-images.githubusercontent.com/3303/78281057-56b95a80-74d7-11ea-82ca-be40e1afa9ab.png)

Reply:
> Are you saying that the resemblance of the first is more natural than the second?

> * (A x, B y) ~ y^* (B^* A) x

Honestly, to me, yes, because I know that in the mathematical definition of inner product for  complex-valued functions/vectors, the order needs to be inverted as also wiki reports for the [Conjugate symmetry property](https://en.wikipedia.org/wiki/Inner_product_space#Definition), or for the [Complex coordinate space example](https://en.wikipedia.org/wiki/Inner_product_space#Complex_coordinate_space). I believe students are more likely to find classical mathematical definitions than more field-specific ones. And they can relate to similar properties like the transpose of the product of two matrices, or the conjugate transpose of the product of two matrices and so on..

Reply:
I won't argue the point further, though perhaps other people would like to comment. The notation in which order is preserved `(v, u) ~ v^T M u` is not niche, just a reflection of how people are introduced to finite elements. I honestly don't know which is more common, and more importantly, the integrated transition cost between this and other notational differences. I often eschew the inner product notation due to this ambiguity, preferring to write `\int \bar v u` or whatever. Of course dropping the `d V` helps make the `\int` notation lighter.

Reply:
I just answered your question on which one looks more natural to me personally, but of course you know what's best for the libCEED's community, and we can revert that.

Reply:
Fair enough. I'd prefer to either leave it or get a few more opinions.